### PR TITLE
Starter Pack: expand scope to all admin pages

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/woocommerce/woocommerce-admin/issues/5608

SlotFill wasn't working with my example because the script was never loaded outside wc-admin pages, but should be loaded on all admin pages. This PR changes that to prevent future headache for extensions using the starter pack to hook into the Nav.

### Detailed test instructions:

1. Just make sure the change makes sense. Any extension using the starter pack will have their scripts load on all admin pages, not just wc-admin. This is important because the new Nav operates outside of just wc-admin pages.

### Changelog Note:

- tweak: Expand default script loading in Create Extension to all admin pages

### CODE REVIEW

File: bin/starter-pack/_main.php [hunk 1]
```diff
@@ -9,11 +9,10 @@
  * Register the JS.
  */
 function add_extension_register_script() {
-
-	if ( ! class_exists( 'Automattic\WooCommerce\Admin\Loader' ) || ! \Automattic\WooCommerce\Admin\Loader::is_admin_page() ) {
+	if ( ! is_admin() ) {
```
Comment (joshuatf) [inline bin/starter-pack/_main.php:12]: Not completely opposed to opening this up for all pages, but could we use `is_admin_or_embed_page()` to allow the nav slot fill to work where needed?

**Review by joshuatf — COMMENTED** [review]
Good call 👍 

Quick question on a possible way to limit loading on every page.

File: bin/starter-pack/_main.php (hunk 1 above)
Comment (psealock) [reply bin/starter-pack/_main.php:12]: oh yeah good idea. Can we be sure `is_admin_or_embed_page` is available to extensions? It isn't in my initial testing.

Comment (joshuatf) [reply bin/starter-pack/_main.php:12]: If they register a page with the new navigation, they will qualify under `is_admin_or_embed_page()`.  Do you have an example extension that doesn't pass this test?  If so, we can keep `is_admin()`.

Comment (psealock) [reply bin/starter-pack/_main.php]: Gotcha, thanks. I updated the code.

**Review by adrianduffell — APPROVED** [review]
This code change makes sense 🚀

**Pull request merged**